src/controllers/user.js

```diff
@@ -12,7 +15,62 @@ const getProfile = async (req, res) => {
         .json(currentUser);
     }
   } catch (error) {
-    return res.sendStatus(500).json({ message: error.message });
+    return res.status(500).json({ message: error.message });
+  }
+};
+
+const getUserProducts = async (req, res) => {
+  try {
+    let { postType, SearchTitles } = req.query;
+    const filteringQueries = { $or: [] };
+    const finalQuery = { $and: [] };
+
+    if (postType) {
+      if (typeof postType !== 'object') postType = postType.split();
+
+      postType.forEach((type) => {
+        if (type === 'Donated') {
+          filteringQueries.$or.push({
+            publisher: req.params.userId,
+            postType: 'Donate',
+          });
+        } else if (type === 'Requested') {
+          filteringQueries.$or.push({
+            publisher: req.params.userId,
+            postType: 'Request',
+            isTransactionCompleted: false,
+          });
+        } else if (type === 'Received') {
+          filteringQueries.$or.push({
+            beneficiary: req.params.userId,
+            postType: 'Request',
+            isTransactionCompleted: true,
+          });
+        }
+      });
+
+      if (filteringQueries.$or.length !== 0) {
+        finalQuery.$and.push(filteringQueries);
+      }
+      if (SearchTitles) {
+        finalQuery.$and.push({
+          title: { $regex: SearchTitles, $options: 'i' },
+        });
+      }
+    }
+
+    if (finalQuery.$and.length === 0) {
+      const userPosts = await ProductModel.find({
+        publisher: req.params.userId,
+      });
+
+      return res.json(userPosts);
```

Comment:
kankam shouldn't you add status code here also?

Reply:
I'll write them and push again